package.json

```diff
@@ -3,39 +3,50 @@
   "private": true,
   "type": "module",
   "scripts": {
-    "dev": "vite",
-    "build": "vite build",
-    "preview": "vite preview",
+    "start": "run-s build:* start:server",
+    "build:client": "vite build",
+    "build:server": "tsc --project tsconfig.server.json",
+    "start:server": "node dist/server/src/server/server.js",
+    "dev": "run-p dev:* start:devServer",
+    "dev:watchClient": "vite build --watch",
+    "dev:watchServer": "tsc --project tsconfig.server.json --watch",
```

Comment:
This not a comment on the code, but just documenting a technique I used to test the "watch" mode for the vite build , tsc, and nodemon.  I used the `touch` shell command to change the date of various files and folders to see what triggered a rebuild and restart.  For the record:

First, the `dist` directory tree:

- `touch dist/client/assets/index-DuP7TeSo.js` caused `nodemon` to restart.
- `touch dist/client/assets/index-DuP7TeSo.css` or the `.map` did not cause `nodemon` to restart
- `touch dist/client/assts/index-DuP7TeSo.js` caused `nodemon` to restart.
- `touch dist/server/config/config.js` caused `nodemon` to restart.
- `touch dist/server/src/server/server.js` caused `nodemon` to restart.
- `touch dist/client` did nothing.
- `touch dist/client/assets` did nothing.
- `touch dist/server/config/` did nothing.
- `touch dist/server/src/` did nothing.

Next, using the same technique in the `src` directory tree:

- `touch src` did nothing.
- `touch src/client` did nothing.
- `touch src/client/ActionBmwCodeCell.scss` caused `vite build` to run. In  turn, this caused `nodemon` to restart (multiple times).
- `touch src/server/server.ts` **did nothing**.

The last one was surprising.  In order to trigger the `tsc` script even when in "watch" mode, it was necessary to change something in the file and save it.  That is different from the `vite build` script which does react to a change in the file's date.